pytorch_lightning/core/optimizer.py

```diff
@@ -116,34 +102,24 @@ def _to_lightning_optimizer(cls, optimizer, trainer, opt_idx):
             optimizer = trainer.lightning_optimizers[opt_idx]
         return optimizer
 
-    def _accumulated_batches_reached(self):
-        if self.accumulate_grad_batches is None:
-            return self._trainer.train_loop._accumulated_batches_reached()
-        return (self._trainer.batch_idx + 1) % self.accumulate_grad_batches == 0
-
-    @property
-    def _should_accumulate(self):
-        # checks if backward or backward + optimizer step (via closure)
-        accumulation_done = self._accumulated_batches_reached()
-        is_final_batch = self._trainer.train_loop._num_training_batches_reached()
-        return not (accumulation_done or is_final_batch)
-
-    def __optimizer_step(self, closure: Optional[Callable] = None, profiler_name: str = None, **kwargs):
+    def __optimizer_step(self, zero_grad: bool = False, closure: Optional[Callable] = None, profiler_name: str = None, **kwargs):
```

Comment:
can you make the closure the first argument, that just specifying it as arg still works?